packages/cydran/src/log/strategy/AbstractLevelStrategy.ts

```diff
@@ -0,0 +1,150 @@
+import { JSType } from "CydranConstants";
+import Level from "log/Level";
+import { Appender } from "log/appender/Appender";
+import { LevelStrategy } from "log/strategy/LevelStrategy";
+import { compositeArray, isDefined } from "util/Utils";
+
+const withStack = (...moreArgs: any): Error | undefined => {
+	const wkps: any[] = moreArgs.length > 0 ? [... moreArgs] : [];
+	const forStackOut: Error | null = (wkps.length > 0 &&  wkps[wkps.length] instanceof Error) ? wkps[wkps.length] : null;
+	return forStackOut;
+}
+
+const resolvemsgBase = (payloadFn: () => string, logLabel: string, pender: Appender): string => {
+	try {
+		return payloadFn();
+	} catch (ex: any) {
+		pender.error(logLabel, "issue resolving log msg:", ex);
+		return "";
+	}
+}
+
+type LogParts = {e2send: Error | null, msg2Log: string, wkArgs: any[]};
+
+export abstract class AbstractLevelStrategy implements LevelStrategy {
+	
+	abstract getLevel(): string;
+
+	public ifTrace(logLabel: string, pender: Appender, payloadFn: () => any, moreArgs: any[]): void {
+		if (this.isTrace()) {
```

Comment:
If conditional not needed, as the level logic is inherent in the strategy implementation in active use

Reply:
fixed